gravitee-alert-engine-connector-ws/src/main/java/io/gravitee/ae/connector/ws/configuration/ConnectorConfiguration.java

```diff
@@ -210,17 +244,17 @@ private Map<String, Engine> initializeEngines() {
         ((AbstractEnvironment) environment).getPropertySources()
             .stream()
             .collect(Collectors.toList())
+            .stream()
+            .filter(propertySource -> propertySource instanceof MapPropertySource)
             .forEach(propertySource -> {
-                if (propertySource instanceof MapPropertySource) {
-                    ((MapPropertySource) propertySource).getSource()
-                        .forEach((k, v) -> {
-                            if (k.startsWith(keyInitial)) {
-                                String replace = k.replace(keyInitial, "");
-                                String name = replace.substring(0, replace.indexOf("."));
-                                engineNames.add(name);
-                            }
-                        });
-                }
+                ((MapPropertySource) propertySource).getSource()
+                    .forEach((k, v) -> {
+                        if (k.startsWith(keyInitial)) {
+                            String replace = k.replace(keyInitial, "");
+                            String name = replace.substring(0, replace.indexOf("."));
+                            engineNames.add(name);
+                        }
+                    });
```

Comment:
The first `stream().collect(Collectors.toList())` seems unneeded.

This is my proposal, with a little refactoring in the `initializeEngines` function :

```suggestion
private Map<String, Engine> initializeEngines() {
        String keyInitial = "alerts.alert-engine.engines.";

        // try first with engines tag
        var result =
            ((AbstractEnvironment) environment).getPropertySources()
                .stream()
                .filter(propertySource -> propertySource instanceof MapPropertySource)
                .flatMap(propertySource -> ((MapPropertySource) propertySource).getSource().keySet().stream())
                .filter(key -> key.startsWith(keyInitial))
                .map(key -> key.replace(keyInitial, "").split("\\.", 1)[0])
                .map(engineName -> {
                    String key = String.format("%s%s", keyInitial, engineName);
                    List<Endpoint> endpointList = initializeEndpoints(key + ".endpoints")
                        .stream()
                        .map(Endpoint::new)
                        .collect(Collectors.toList());
                    String uname = environment.getProperty(key + ".security.username");
                    String pass = environment.getProperty(key + ".security.password");
                    return Map.entry(engineName, new Engine(this, endpointList, new Engine.Security(uname, pass)));
                })
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));

        if (!result.isEmpty()) {
            if (result.get(DEFAULT_ENGINE_NAME) == null) {
                throw new RuntimeException("Default engine is not found! You need to have a default engine in your configuration.");
            }
        } else { // for backward compatibility
            Engine defaultEngine = new Engine(
                this,
                initializeEndpoints("alerts.alert-engine.ws.endpoints").stream().map(Endpoint::new).collect(Collectors.toList()),
                new Engine.Security(this.getUsername(), this.getPassword())
            );

            result.put(DEFAULT_ENGINE_NAME, defaultEngine);
        }

        return result;
    }
```

I run a `mvn clean verify` successfully.

Reply:
I'd rather not touch this right now as we have something coming with multiple DC connector which may impact this.
This could be a good opportunity to do it there.

Good catch 👍